compiler/rustc_query_system/src/ich/mod.rs

```diff
@@ -17,3 +18,124 @@ pub const IGNORED_ATTRIBUTES: &[Symbol] = &[
     sym::rustc_partition_codegened,
     sym::rustc_expected_cgu_reuse,
 ];
+
+#[allow(rustc::usage_of_ty_tykind)]
+impl<'__ctx, I: rustc_type_ir::Interner> HashStable<StableHashingContext<'__ctx>>
```

Comment:
I think this, along with the other "should be derives but we can't" impls, deserves to have a comment of why we're essentially replicating `derive`.

Reply:
Ah yes, can do.